python/python/glide/async_commands/core.py

```diff
@@ -670,6 +670,84 @@ async def hkeys(self, key: str) -> List[str]:
         """
         return cast(List[str], await self._execute_command(RequestType.Hkeys, [key]))
 
+    async def hrandfield(self, key: str) -> Optional[str]:
+        """
+        Returns a random field name from the hash value stored at `key`.
+
+        See https://redis.io/commands/hrandfield/ for more details.
+
+        Args:
+            key (str): The key of the hash.
+
+        Returns:
+            Optional[str]: A random field name from the hash stored at `key`.
+            If the hash does not exist or is empty, None will be returned.
+
+        Examples:
+            >>> await client.hrandfield("my_hash")
+                "field1"  # A random field name stored in the hash "my_hash".
+        """
+        return cast(
+            Optional[str], await self._execute_command(RequestType.HRandField, [key])
+        )
+
+    async def hrandfield_count(self, key: str, count: int) -> List[str]:
+        """
+        Retrieves random field names from the hash value stored at `key`.
+
+        See https://redis.io/commands/hrandfield/ for more details.
+
+        Args:
+            key (str): The key of the hash.
+            count (int): The number of field names to return.
+                If `count` is positive, returns unique elements.
+                If negative, allows for duplicates.
+
+        Returns:
+            List[str]: A list of random field names from the hash.
+            If the hash does not exist or is empty, the response will be an empty list.
+
+        Examples:
+            >>> await client.hrandfield_count("my_hash", -3)
+                ["field1", "field1", "field2"]  # Non-distinct, random field names stored in the hash "my_hash".
+            >>> await client.hrandfield_count("non_existing_hash", 3)
+                []  # Empty list
+        """
+        return cast(
+            List[str],
+            await self._execute_command(RequestType.HRandField, [key, str(count)]),
+        )
+
+    async def hrandfield_count_withvalues(
+        self, key: str, count: int
+    ) -> List[List[str]]:
```

Comment:
Should return a map I think - data should be converted in glide core - search for `ExpectedReturnType::MapOfStringToDouble` for example

Reply:
The problem with using a map is that a negative count value means fields can be repeated in the redis response (in other words, fields can be returned more than once in the list). If it were a map we wouldn't be able to represent repeated fields

Reply:
Oh right
But you still need data conversion to be done in glide to ensure the same format in resp2/3
```
$ redis-cli -3
127.0.0.1:6379> HRANDFIELD coin -5 WITHVALUES
1) 1) "edge"
   2) "null"
2) 1) "edge"
   2) "null"
3) 1) "edge"
   2) "null"
4) 1) "tails"
   2) "reverse"
5) 1) "edge"
   2) "null"
```
```
$ redis-cli -2
127.0.0.1:6379> HRANDFIELD coin -5 WITHVALUES
 1) "heads"
 2) "obverse"
 3) "heads"
 4) "obverse"
 5) "edge"
 6) "null"
 7) "heads"
 8) "obverse"
 9) "heads"
10) "obverse"
```

Reply:
Yes that is one problem I've noticed that I am not sure how to address. If resp3 returns List[List[Str]] and resp2 returns List[str], what should the signature of the function return?

Reply:
From our slack conversation sounds like we are waiting on AWS approval for what type to return

Reply:
We are now approved.  Please go ahead and add the data conversion.